vault/identity_lookup.go

```diff
@@ -36,6 +59,40 @@ func lookupPaths(i *IdentityStore) []*framework.Path {
 	}
 }
 
+func (i *IdentityStore) pathLookupEntityUpdate(req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+	lookupType := d.Get("type").(string)
+	if lookupType == "" {
+		return logical.ErrorResponse("empty type"), nil
+	}
+
+	switch lookupType {
+	case "id":
+		entityID := d.Get("id").(string)
+		if entityID == "" {
+			return logical.ErrorResponse("empty id"), nil
+		}
+		entity, err := i.memDBEntityByID(entityID, false)
+		if err != nil {
+			return nil, err
+		}
+		return i.handleEntityReadCommon(entity)
+	case "name":
+		entityName := d.Get("name").(string)
+		if entityName == "" {
+			return logical.ErrorResponse("empty name"), nil
+		}
+		entity, err := i.memDBEntityByName(entityName, false)
+		if err != nil {
+			return nil, err
+		}
+		return i.handleEntityReadCommon(entity)
```

Comment:
This, and the call above, could be moved to the end of the function to avoid calling it twice.

Reply:
Fixed.